cpp/dolfinx/io/XDMFFile.cpp

```diff
@@ -336,8 +337,28 @@ XDMFFile::read_meshtags(const mesh::Mesh<double>& mesh, std::string name,
 
   const auto [entities, eshape] = read_topology_data(name, xpath);
 
-  pugi::xml_node values_data_node
-      = grid_node.child("Attribute").child("DataItem");
+  pugi::xml_node attribute_node = grid_node.child("Attribute");
+  pugi::xml_node values_data_node = attribute_node.child("DataItem");
+  if (!attribute_label.empty())
+  {
+    bool found = false;
+    while (!found and attribute_node)
```

Comment:
I remember looking at this. The problem was, if I recall correctly, that I am searching for an attribute by the name of `"Name"` and the "basic functionality" functions do not allow to search for a name. If I wanted to loop using `next_attribute` I would then need to check if the attribute's name is `"Name"` and, if not, `continue`.

I will add documentation but the code doesn't look to exotic to me [because I wrote it, of course XD].